dbt State dev environment and user credentials (#9433)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/dbt-state-setup.md

4. In the **Execution settings** section of the job, select **Enable dbt State**.
5. Click **Save**.

> #### Enabling dbt State in the Studio IDE

Reviewer: not a blocker but sugg a shorter header
```suggestion
#### Enabling dbt State in Studio
```

> dbt State can run automatically while you work in your <Constant name="studio_ide" />. You can [turn it on for your development environment](#enabling-dbt-state-on-a-development-environment) so it's the default for everyone, or you can [override that setting just for your own account](#overriding-dbt-state-setting-per-user).

Reviewer: i'm wondering if we're missing the 'why' here? like what can i do wit this? i know we mention this earlier in the page but i think it's helpful to have a reminder
```suggestion
When you enable dbt State in the <Constant name="studio_ide" />, it runs automatically on every `dbt run` or `dbt build` during development &mdash; skipping unchanged models and reusing production results so your runs are _faster_.

You can [turn it on for your development environment](#enabling-dbt-state-on-a-development-environment) so it's the default for everyone, or you can [override that setting just for your own account](#overriding-dbt-state-setting-per-user).
```

> **Prerequisite**: dbt State must be [enabled on your account](#enabling-dbt-state-on-your-account).

Author: @peter-bertuglia is it correct that this is a prerequisite for both enabling dbt state on a dev environment and overriding the configuration per user?

I looked at the user credentials field with dbt State disabled in the account -- it's there and I can select a value but I don't think enabling it in this field enabled dbt State in my IDE

Reviewer 2: i'll double check that. it should be a prerequisite that dbt State is enabled for the account to enable dbt state in any way

Reviewer 2: i've got a PR up to fix this, thanks for flagging! https://github.com/dbt-labs/cloud-ui/pull/6980

> **Prerequisite**: dbt State must be [enabled on your account](#enabling-dbt-state-on-your-account).

Reviewer: sugg active voice
```suggestion
**Prerequisite**: An account admin must [enable dbt State](#enabling-dbt-state-on-your-account) before you can use it.
```

##### Enabling dbt State on a development environment

Enabling dbt State on your development environment turns it on for everyone using the <Constant name="studio_ide" />, unless they override it for their own account.

1. Go to **Orchestration** > **Environments** and select your development environment.
2. Click **Settings** > **Edit**.
3. In the **dbt State** section, select **Enable dbt State**.

> 4. Click **Save**.

Reviewer: sugg this extra step since it's prob important for users to know that they'll get an ide restart:
<img width="716" height="319" alt="Screenshot 2026-06-17 at 11 33 46" src="https://github.com/user-attachments/assets/357e0796-bea9-4a8d-9be7-a7b388cc6c9f" />

```suggestion
4. Click **Save**. 
5. In the pop-up box, click **Continue** if you want to go ahead with the changes and restart all IDE sessions for this project.
```

##### Overriding dbt State setting per user

You can override the development environment's dbt State setting for your own account without affecting other users. Because the user-level setting takes precedence over the environment-level setting, you can turn dbt State on for yourself before enabling it for your whole team, or turn it off when it's enabled at the environment level.

1. Click your account name in the lower-left corner and select **Account settings**.
2. Under **Your profile**, go to **Credentials**.
3. Select the project you want to enable dbt State for.
4. Click **Edit** and go to the **User development settings** section.
5. Under **dbt State**, select one of the following options:

>    - **Enabled** — Enables dbt State for your user regardless of the development environment setting.

Reviewer: ```suggestion
   - **Enabled**: Enables dbt State for your user regardless of the development environment setting.
```

>    - **Disabled** — Disables dbt State for your user regardless of the development environment setting.

Reviewer: ```suggestion
   - **Disabled**: Disables dbt State for your user regardless of the development environment setting.
```

>    - **Reset (inherit from development)** — Clears your override and falls back to the dbt State setting configured on your development environment. This option only appears after you've saved an **Enabled** or **Disabled** override.

Reviewer: ```suggestion
   - **Reset (inherit from development)**: Only appears after you've saved an **Enabled** or **Disabled** override. Clears your override and falls back to the dbt State setting configured on your development environment.
```
